dm/worker/server.go

```diff
@@ -749,3 +780,85 @@ func makeCommonWorkerResponse(reqErr error) *pb.CommonWorkerResponse {
 	}
 	return resp
 }
+
+// all subTask in subTaskCfgs should have same source
+// this function return the min position in all subtasks, used for relay's position
+func getMinPosInAllSubTasks(subTaskCfgs []*config.SubTaskConfig) (minPos *mysql.Position, err error) {
+	for _, subTaskCfg := range subTaskCfgs {
+		pos, err := getMinPosForSubTaskFunc(subTaskCfg)
+		if err != nil {
+			return nil, err
+		}
+
+		if pos == nil {
+			continue
+		}
+
+		if minPos == nil {
+			minPos = pos
+		} else {
+			if minPos.Compare(*pos) >= 1 {
+				minPos = pos
+			}
+		}
+	}
+
+	return minPos, nil
+}
+
+func getMinPosForSubTask(subTaskCfg *config.SubTaskConfig) (minPos *mysql.Position, err error) {
```

Comment:
This function depends on the internal implementation of Checkpoint. How about adding such a method in Checkpoint?

Reply:
update in 69363d7